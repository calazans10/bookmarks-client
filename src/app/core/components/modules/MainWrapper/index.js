import React from 'react';
import PropTypes from 'prop-types';
import Alert from '../../../../ui/components/modules/Alert';
import { Container } from './styles';

const MainWrapper = ({ children }) => (
  <Container>
    <Alert />
    {children}
  </Container>
);

MainWrapper.propTypes = {
  children: PropTypes.node.isRequired,
};

export default MainWrapper;
