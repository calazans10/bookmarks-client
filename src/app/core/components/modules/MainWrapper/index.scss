@import 'styles/utilities';

.main-wrapper {
  position: relative;
  width: 100%;
}
