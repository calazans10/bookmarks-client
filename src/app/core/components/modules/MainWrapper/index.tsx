import React from 'react';
import Alert from 'app/ui/components/modules/Alert';
import { Container } from './style';

type MainWrapperProps = {
  children: React.ReactNode;
};

const MainWrapper = ({ children }: MainWrapperProps) => (
  <Container>
    <Alert />
    {children}
  </Container>
);

export default MainWrapper;
