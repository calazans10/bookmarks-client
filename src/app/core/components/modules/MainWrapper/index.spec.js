import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MainWrapper from './index';

describe('MainWrapper', () => {
  it('renders without crashing', () => {
    const props = {
      children: <p>Lorem Ipsum</p>,
    };
    const wrapper = shallow(<MainWrapper {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
