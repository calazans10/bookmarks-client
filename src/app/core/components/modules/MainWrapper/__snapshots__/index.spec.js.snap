// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainWrapper renders without crashing 1`] = `
<styled.div>
  <Connect(Alert) />
  <p>
    Lorem Ipsum
  </p>
</styled.div>
`;
